src/components/users/UsersIndex.js

```diff
@@ -0,0 +1,77 @@
+import React, { Component, useReducer } from 'react';
+import Table from '../Table';
+
+// library for AJAX functions
+const axios = require('axios');
+
+// This component is a class because it has state. Having state is neccesary
+// because the page will render before the AJAX is complete. Once the AJAX
+// is processed and returned from backend we setState() which will automatically
+// rerender the component with the new state. An additional reason this
+// componenet has state is because if AJAX fails, no results will be available
+// to render
+class UsersIndex extends Component {
+  constructor(props) {
+    super(props);
+
+    // initial state which will get updated from AJAX
+    this.state = {
+      index: []
+    };
+    
+    this.getUsers = this.getUsers.bind(this);
+
+  }
+  
+  // after component mounts ( mount is React term for after the initital render)
+  componentDidMount() {
+
+    // call function to send AJAX to backend
+    this.getUsers();
+  }
+
+  getUsers() {
+
+    // AJAX call to /api/user
+    axios.get('/api/user').then(res => {
+      
+      // update state based on return JSON from backend
+      // React will automatically rerender the component after state updates.
+      // Always use setState to update state. Updating state directly 
+      // can cause errors.
+      this.setState({ index: res.data });
+    });
+  }
+
+  // This render function is called when component first renders
+  // e.g. user clicks on link or when setState updates state
+  render() {
+
+    // set index from state
+    const { index } = this.state
+
+    // set values as empty arrays for initial render or AJAX fail
+    let tableHeaders = [];
+    let tableRows = [];
```

Comment:
Why not `const`?

Reply:
in lines 59-60 I am reassigning them if the AJAX call was succesful. Is there a better way?

Reply:
I don't think this even needs reassigning, you could do something like: 
```
const tableHeaders = index.length ? Object.keys(index[0]) : []
```

And why reassign `index` to `tableRows` when both are arrays and you'll be using both the same way?

Reply:
Thanks I always forget about ternary operators haha.

Reply:
Yea I changed tableRows to index too. I think I did that cause I was getting use to eslint.